src/tqec/templates/library/spatial.py

```diff
@@ -0,0 +1,430 @@
+"""Provide functions building the :class:`~tqec.templates.rpng.RPNGTemplate`
+instances representing spatial junctions."""
+
+from typing import Literal
+
+from tqec.compile.specs.enums import JunctionArms
+from tqec.exceptions import TQECException
+from tqec.plaquette.enums import MeasurementBasis, ResetBasis
+from tqec.plaquette.frozendefaultdict import FrozenDefaultDict
+from tqec.plaquette.rpng import RPNGDescription
+from tqec.templates.indices.qubit import (
+    QubitHorizontalBorders,
+    QubitSpatialJunctionTemplate,
+    QubitVerticalBorders,
+)
+from tqec.templates.rpng import RPNGTemplate
+
+
+def get_spatial_junction_qubit_template(
+    external_stabilizers: Literal["x", "z"],
+    arms: JunctionArms,
+    reset: ResetBasis | None = None,
+    measurement: MeasurementBasis | None = None,
+) -> RPNGTemplate:
+    """Implementation of a logical qubit performing a spatial junction.
+
+    This function returns a RPNGTemplate instance representing a logical qubit
+    that is touched by 2 or more spatial junctions. The returned template is
+    carefully crafted to avoid hook errors damaging the logical distance.
+
+    Note:
+        this function does not enforce anything on the input values. As such, it
+        is possible to generate a description of a round that will both reset and
+        measure the data-qubits.
+
+    Warning:
+        By convention, this function does not populate the plaquettes on the
+        boundaries where an arm (i.e. spatial junction) is present **BUT** do
+        populate the corners (that are part of the boundaries, so this is an
+        exception to the first part of the sentence).
+
+        The rationale behind that convention is that the logical qubit at the
+        center of the spatial junction is completely aware of all the spatial
+        junctions that should be implemented whereas each spatial junction in
+        isolation does not know if the others spatial junctions are present or
+        not. That means that corners, whose plaquette depends on the presence or
```

Comment:
In my view, a spatial junction refers to the logical qubit located at the center of an "X," "T," or "L"-shaped structure. The terminology used here in the documentation is not clear to me.

Reply:
I agree. The terminology I will use from now on is "spatial junction" for the logical qubit performing the junction and "arm" for the pipes around the spatial junction. Does that seems reasonable to you?